velox/functions/lib/Re2Functions.cpp

```diff
@@ -420,22 +421,50 @@ bool matchSubstringPattern(
       std::string::npos);
 }
 
+// Return true if the input VARCHAR argument is all-ASCII for the specified
+// rows.
+static bool isAsciiArg(const SelectivityVector& rows, const VectorPtr& arg) {
+  VELOX_DCHECK(
+      arg->type()->isVarchar(), "Input vector is expected to be VARCHAR type.");
+
+  auto stringArg = arg->asUnchecked<SimpleVector<StringView>>();
+  auto isAscii = stringArg->isAscii(rows);
+
+  if (isAscii.has_value()) {
+    return isAscii.value();
+  } else {
+    // isAscii info is not computed yet, compute it on the fly.
+    return stringArg->computeAndSetIsAscii(rows);
```

Comment:
Looks like we can just call computeAndSetIsAscii and it will either return right away (from cached metadata) or compute the value, update the cache and return. CC: @kgpai